recipes/knotinframe/meta.yaml

```diff
@@ -1,37 +1,42 @@
 {% set version = "2.3.1" %}
-{% set sha256 = "d9ab68329e48709cffbf5ed775f905bdddebab470d70836bb522993b971d48af" %}
+{% set packagename = "knotinframe" %}
+{% set sha256 = "0c5b7d91893d0a23189f4af3602e18f110f030eac49a3658f7acede23596fde0" %}
 
 package:
-  name: knotinframe
-  version: {{ version }}
+  name: {{ packagename }}
+  version: 2.2.0
 
 source:
   sha256: {{ sha256 }}
+  # the repository "fold-grammars" contains sources for several programs, i.e. pKiss, RNAshapes, ...
+  # thus, the version number does not correspond to the {{ packagename }} version number
   url: https://github.com/jlab/fold-grammars/archive/{{ version }}.tar.gz
 
 build:
-  number: 1
+  number: 0
   run_exports:
     - {{ pin_subpackage('knotinframe', max_pin="x") }}
 
 requirements:
   build:
     - make
     - {{ compiler('cxx') }}
+    - libcxx <17 # [osx]
   host:
-    - bellmans-gapc >=2020.12.08
+    - perl
+    - bellmans-gapc >=2024.01.12
+    - libcxx <17 # [osx]
   run:
     - perl
-    - bellmans-gapc >=2020.12.08
+    - bellmans-gapc >=2024.01.12
 
 test:
   commands:
   - echo "AUGC" | knotinframe | grep 'No suitable slippery sites'
 
 about:
-  home: https://bibiserv.cebitec.uni-bielefeld.de/knotinframe
+  home: https://bibiserv.cebitec.uni-bielefeld.de/{{ packagename }}
```

Comment:
Replacing `knotinframe` with a template `packagename` would make sense if you were building multiple tools from that location in one recipe, which is not what we usually do. Since you aren't using the template throughout, nor building multiple things here, I'm not sure why you are using a template for the URL.

Reply:
because I was using copy & paste from the above mentioned other packages like https://github.com/bioconda/bioconda-recipes/blob/f20dff4c1331bd49d80d3500eacad472a1212b91/recipes/rnaalishapes/meta.yaml#L41 or https://github.com/bioconda/bioconda-recipes/blob/f20dff4c1331bd49d80d3500eacad472a1212b91/recipes/palikiss/meta.yaml#L41